src/main/java/edu/greenblitz/tobyDetermined/subsystems/swerve/Modules/SimulationSwerveModule.java

```diff
@@ -0,0 +1,115 @@
+                DCMotor.getFalcon500(RobotMap.Swerve.Simulation.SIMULATION_LINEAR_MOTOR.NUMBER_OF_MOTORS),
+                RobotMap.Swerve.Simulation.SIMULATION_LINEAR_MOTOR.GEAR_RATIO,
+                RobotMap.Swerve.Simulation.SIMULATION_LINEAR_MOTOR.MOMENT_OF_INERTIA
+                );
+        this.angularMotor = new DCMotorSim(
+                DCMotor.getFalcon500(RobotMap.Swerve.Simulation.SIMULATION_ANGULAR_MOTOR.NUMBER_OF_MOTORS),
+                RobotMap.Swerve.Simulation.SIMULATION_ANGULAR_MOTOR.GEAR_RATIO,
+                RobotMap.Swerve.Simulation.SIMULATION_ANGULAR_MOTOR.MOMENT_OF_INERTIA
+        );
+    }
+
+
+    @Override
+    public void setLinearVelocity(double speed) {
+        final double power = speed / RobotMap.Swerve.MAX_VELOCITY;
+        final double voltage = power * RobotMap.SimulationConstants.BATTERY_VOLTAGE;
+        setLinearVoltage(voltage);
+    }
+
+    @Override
+    public void rotateToAngle(double angleInRadians) {
+        double diff = Math.IEEEremainder(angleInRadians - lastInputs.angularPositionInRads, 2 * Math.PI);
+        diff -= diff > Math.PI ? 2 * Math.PI : 0;
+        angleInRadians = lastInputs.angularPositionInRads + diff;
+
+        angularController.setSetpoint(angleInRadians);
+        final double voltage = angularController.calculate(lastInputs.angularPositionInRads);
+        setAngularVoltage(voltage);
+    }
+
+    @Override
+    public void setLinearVoltage(double voltage) {
+        linearAppliedVoltage = voltage;
+        linearMotor.setInputVoltage(voltage);
+    }
+
+    @Override
+    public void setAngularVoltage(double voltage) {
+        angularAppliedVoltage = voltage;
+        angularMotor.setInputVoltage(voltage);
+    }
+    @Override
+    public void stop() {
+        setLinearVoltage(0);
+        setAngularVoltage(0);
+    }
+
+    @Override
+    public void updateInputs(SwerveModuleInputsAutoLogged inputs) {
+        linearMotor.update(RobotMap.SimulationConstants.TIME_STEP);
+        angularMotor.update(RobotMap.SimulationConstants.TIME_STEP);
+
+        inputs.linearVelocity = Conversions.MK4IConversions.convertRPMToMeterPerSecond(linearMotor.getAngularVelocityRPM()); // [m/s]
+        inputs.angularVelocity = angularMotor.getAngularVelocityRadPerSec(); // [Rad/s]
+
+        inputs.linearVoltage = linearAppliedVoltage;
+        inputs.angularVoltage = angularAppliedVoltage;
+
+        inputs.linearCurrent = linearMotor.getCurrentDrawAmps();
+        inputs.angularCurrent = angularMotor.getCurrentDrawAmps();
+
+        inputs.linearMetersPassed = Conversions.MK4IConversions.convertRevolutionToMeters(linearMotor.getAngularPositionRotations());
+        inputs.angularPositionInRads = angularMotor.getAngularPositionRad();
+
+        inputs.absoluteEncoderPosition = inputs.angularPositionInRads;
+        inputs.isAbsoluteEncoderConnected = true;
+
+        lastInputs = inputs;
+    }
+
+
+    @Override
+    public void setLinearIdleModeBrake(boolean isBrake) {
```

Comment:
Probably would want to log a use of those functions (also setAngulrIdleModeBreak, resetAngle)

Reply:
can it be prints? you can see the console in the advantage scope

Reply:
now recording the output